src/peft/tuners/tuners_utils.py

```diff
@@ -814,11 +814,23 @@ def _maybe_include_all_linear_layers(peft_config: PeftConfig, model: nn.Module)
             names = name.rsplit(".", 1)[-1]  # get the base name
             linear_module_names.add(names)
 
-    # ignore the last classification head for text generation models
+    # Try to remove linear layers that should not be targeted as best as possible. We have to rely on convention as
+    # there are no hard rules to detect these modules.
+    module_names_to_exclude = set()
     output_emb = model.get_output_embeddings()
     if output_emb is not None:
+        # ignore the last classification head for text generation models
         last_module_name = [name for name, module in model.named_modules() if module is output_emb][0]
-        linear_module_names -= {last_module_name}
+        module_names_to_exclude.add(last_module_name)
+    elif peft_config.task_type == TaskType.SEQ_CLS:
+        # ignore classifier head for classification models (issue 2027)
+        # there is no fix name for the classifier head, "score" and "classifier" are most common
+        cls_head = getattr(model, "score", None) or getattr(model, "classifier", None)
+        if cls_head is not None:
+            last_module_name = [name for name, module in model.named_modules() if module is cls_head][0]
+            module_names_to_exclude.add(last_module_name)
```

Comment:
Perhaps we could define a MAP between the task types and the attributes we know we should exclude and use that? 

```python
EXCLUSION_MAP = {TaskType.SEQ_CLS: ["score", "classifier"], ...}

...

cls_head = None
for exclude_candidate in EXLCUSION_MAP[TaskType.SEQ_CLS]:
    cls_head = getattr(model, exclude_candidate, None)

if cls_head is not None:
    ...
```

The advantage of that is we just have to update the MAP in case we discover more attrbiutes and it should work out nicely. 

WDYT?

Reply:
Good point about making this easier to extend. I'm not sure if a map is the right approach, because for causal LM (the `if` condition above), we use a different approach based on `get_output_embeddings()`, so the map could not be used consistently for that task. But I will move `["score", "classifier"]` into a constant and use that.

Reply:
I still think a map or any adjacent approach is better in the long term but you of course know better here.